src-ui/sw.js

```diff
@@ -0,0 +1,23 @@
+var CACHE_NAME = "v3";
+var urlsToCache = ["/p"];
+
+self.addEventListener("install", function(event) {
+	event.waitUntil(
+		caches.open(CACHE_NAME).then(function(cache) {
+			return cache.addAll(urlsToCache);
+		})
+	);
+});
+
+self.addEventListener("fetch", function(event) {
+	event.respondWith(
+		caches
```

Comment:
to fix the stale cache trouble I've been having, this should either query the latest cache only, or we need to add an activation handler that deletes the old caches